File: InterviewTraining/3sum.py
def threeNumberSum(array, targetSum):
	# Write your code here.
	arrSorted=sorted(array)
	retval = []

	idx1, idx2, idx3=0 , 0, 0
	while idx1 < len(arrSorted)-2:
		val1=arrSorted[idx1]
		idx2 = idx1 + 1
		while idx2 < len(arrSorted)-1:
			val2=arrSorted[idx2] 
			idx3=idx2+1
			while idx3 < len(arrSorted):
				val3=arrSorted[idx3] 
				idx3 += 1
				if val1 + val2 + val3 == targetSum:
					retval.append([val1,val2,val3])
				print(idx1,idx2,idx3)

			idx2 += 1	


			idx1 += 1
			
	return retval

tests = [
	{
  "array": [12, 3, 1, 2, -6, 5, -8, 6],
  "targetSum": 0
	}

]

testid=0
retval = threeNumberSum(tests[testid]["array"],tests[testid]["targetSum"])

print ("retval",retval)

File: InterviewTraining/js/3sum.js
function threeNumberSum(array, targetSum) {
    const arrSorted = array.slice().sort((a, b) => a - b);
    const retval = [];
    for (let i = 0; i < arrSorted.length - 2; i++) {
        for (let j = i + 1; j < arrSorted.length - 1; j++) {
            for (let k = j + 1; k < arrSorted.length; k++) {
                if (arrSorted[i] + arrSorted[j] + arrSorted[k] === targetSum) {
                    retval.push([arrSorted[i], arrSorted[j], arrSorted[k]]);
                }
            }
        }
    }
    return retval;
}

module.exports = threeNumberSum;
